### ISSUE TITLE
Support exiting scope before last field `#[scope(exit_before)]`

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#6311]
`oxc_ast_tools` recognises `#[scope(enter_before)]` to mark a field as the start of the scope in a type. We also need `#[scope(exit_before)]` for #6270.

Note: I think `#[scope(exit_before)]` is better than `#[scope(exit_after)]` because attributes go before fields, so it reads more clearly:

```rs
pub struct TSConditionalType<'a> {
    pub span: Span,
    pub check_type: TSType<'a>,
    #[scope(enter_before)]
    pub extends_type: TSType<'a>,
    pub true_type: TSType<'a>,
    #[scope(exit_before)]
    pub false_type: TSType<'a>,
    pub scope_id: Cell<Option<ScopeId>>,
}
```

We will need to build support for this into both:

1. `oxc_ast_tools` and the `Visit` / `VisitMut` generator.
2. `oxc_traverse`'s codegen.

**Comment by overlookmotel** [comment]
@DonIsaac Rzvxa is the master of the codegen, but is off radar at the moment, and I'm not sure when I'd get to it. You comfortable enough with `oxc_ast_tools` to implement yourself?

**Comment by overlookmotel** [comment]
We could also change `#[scope(enter_before)]` and `#[scope(exit_before)]` to the simpler `#[scope(enter)]` and `#[scope(exit)]`. I don't think "_before" is adding much value. This looks equally clear to me:

```rs
pub struct TSConditionalType<'a> {
    pub span: Span,
    pub check_type: TSType<'a>,
    #[scope(enter)]
    pub extends_type: TSType<'a>,
    pub true_type: TSType<'a>,
    #[scope(exit)]
    pub false_type: TSType<'a>,
    pub scope_id: Cell<Option<ScopeId>>,
}
```

**Comment by DonIsaac** [comment]
I'm not very comfortable, but I can take a look and try it out.

**Comment by overlookmotel** [comment]
Implemented in #6350. Thanks Don!
